packages/flutter_tools/test/widget_preview_scaffold.shard/widget_preview_scaffold/test/filter_by_selected_file_test.dart

```diff
@@ -11,112 +11,143 @@ import 'package:widget_preview_scaffold/src/widget_preview_rendering.dart';
+  dtdServices.selectedSourceFile.value = TextDocument(
+    uriAsString: textEditorScript2Uri,
+    version: 0,
+  );
+  await tester.pumpWidget(widgetPreview);
 
-    // Select kScript2
-    dtdServices.selectedSourceFile.value = TextDocument(
-      uriAsString: kScript2,
-      version: 0,
-    );
-    await tester.pumpWidget(widgetPreview);
+  // Verify only previews from script2Uri are displayed.
+  expect(dtdServices.selectedSourceFile.value?.uriAsString, script2Uri);
+  expect(
+    controller
+        .filteredPreviewSetListenable
+        .value
+        .single
+        .previews
+        .single
+        .scriptUri,
+    script2Uri,
+  );
 
-    // Verify only previews from kScript2 are displayed.
-    expect(dtdServices.selectedSourceFile.value?.uriAsString, kScript2);
-    expect(
-      controller
-          .filteredPreviewSetListenable
-          .value
-          .single
-          .previews
-          .single
-          .scriptUri,
-      kScript2,
-    );
+  final Finder filterBySelectedFileToggle = find.byType(
+    FilterBySelectedFileToggle,
+  );
 
-    final Finder filterBySelectedFileToggle = find.byType(
-      FilterBySelectedFileToggle,
-    );
+  // Press the "Filter by selected file" button and disable preview filtering.
+  expect(controller.filterBySelectedFileListenable.value, true);
+  await tester.tap(filterBySelectedFileToggle);
+  expect(controller.filterBySelectedFileListenable.value, false);
+  // Verify the currently selected source is still script2Uri but all previews are displayed.
+  expect(dtdServices.selectedSourceFile.value?.uriAsString, script2Uri);
+  expect(controller.filteredPreviewSetListenable.value, groups);
+}
 
-    // Press the "Filter by selected file" button and disable preview filtering.
-    expect(controller.filterBySelectedFileListenable.value, true);
-    await tester.tap(filterBySelectedFileToggle);
-    expect(controller.filterBySelectedFileListenable.value, false);
-    // Verify the currently selected source is still kScript2 but all previews are displayed.
-    expect(dtdServices.selectedSourceFile.value?.uriAsString, kScript2);
-    expect(controller.filteredPreviewSetListenable.value, groups);
+void main() {
+  testWidgets('Filter previews based on currently selected file (POSIX)', (
+    tester,
+  ) async {
+    const kScript1 = 'file:///script1';
+    const kScript2 = 'file:///script2';
+    await testImpl(
+      tester: tester,
+      script1Uri: kScript1,
+      script2Uri: kScript2,
+      textEditorScript1Uri: kScript1,
+      textEditorScript2Uri: kScript2,
+    );
+  });
+  testWidgets('Filter previews based on currently selected file (Windows)', (
+    tester,
+  ) async {
+    // Regression test for https://github.com/flutter/flutter/issues/175524
+    const kScript1 = 'file:///C:/script1';
+    const kScript2 = 'file:///C:/script2';
+    const kTextEditorScript1 = 'file:///C%3A/script1';
+    const kTextEditorScript2 = 'file:///C%3A/script2';
```

Comment:
This reminds me that we can also get inconsistent drive letter casing on Windows, so I think we should ensure uppercase and lowercase drive letters are treated the same (actually the entire path could be different case and - depending on the filesystem - might be case-insensitive - but in practice it's only the drive letter that causes us issues and that's all we usually try to handle elsewhere).

I don't know if the decoding could cause any issues, but if so, maybe calling `toFilePath()` (ensuring it's a file scheme) and comparing the path would be better (although it still won't handle the drive letter casing).

Reply:
`toFilePath()` won't handle the decoding either, so I'm not sure doing that would make any difference here.

Do we have Dart code anywhere that you're aware of where we perform this sort of comparison?